### PR TITLE
Fix various float parsing issues

### PR DESCRIPTION
**Pull request by darichey**
### Summary & Motivation
Fixes parsing of expressions like `.5e1` which are valid floats in Nix.

I simply extracted the logic which applies to similar expressions with leading digits (e.g., `1.5e1`) and called it in both places.

Edit: Now also fixes parsing of expresions like `1.` and `1e+3` (scientific notation with a plus).

### CODE REVIEW

**Comment by ncfavier** [comment]
Does this also fix parsing for `1.` ?

**Comment by darichey** [comment]
It does not. I can look into that!

File: src/tokenizer.rs [hunk 1]
```diff
@@ -440,23 +440,27 @@ impl<'a> Tokenizer<'a> {
...
                 }
             }
             _ => TOKEN_ERROR,
         })
     }
+
+    fn consume_scientific(&mut self) -> SyntaxKind {
+        if self.peek() == Some('e') || self.peek() == Some('E') {
+            self.next().unwrap();
+            if self.peek() == Some('-') {
```
Comment (oberblastmeister) [inline src/tokenizer.rs:455]: This can be `+` also

**Comment by darichey** [comment]
@ncfavier @oberblastmeister both of those issues should be fixed now as well!

**Review by ncfavier — APPROVED** [review]

**Pull request merged**